### PR TITLE
[PAY-882] Individual Chat Screen

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description

- Implements individual chat screen for mobile.
- Moves date formatting for chat messages into common util.
- Converts ScrollView to FlatList in ChatListScreen.

Todo:
- first unread marker
- initial scroll location
- message tails
- header shadow
- UserBadges in header
- fetch on scroll

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

ios sim

Simulator Screen Shot - iPhone 14 Pro - 2023-02-09 at 02 18 47

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-PAY-882

File: packages/mobile/src/screens/chat-screen/ChatListScreen.tsx [hunk 1]
```diff
@@ -1,42 +1,39 @@
...
+  rootContainer: {
     display: 'flex',
     justifyContent: 'center',
-    alignItems: 'center',
     height: '100%'
   },
   loadingSpinner: {
     height: spacing(20),
-    widht: spacing(20)
+    widht: spacing(20),
```
Comment (rickyrombo) [inline packages/mobile/src/screens/chat-screen/ChatListScreen.tsx:27]: typo

File: packages/mobile/src/screens/chat-screen/ChatListScreen.tsx [hunk 2]
```diff
@@ -45,17 +42,17 @@ export const ChatListScreen = () => {
...
-        </ScrollView>
+        <View style={styles.rootContainer}>
+          {chatsStatus === Status.SUCCESS ? (
+            <FlatList
+              data={chats}
+              renderItem={({ item, index }) => <ChatListItem chat={item} />}
+              keyExtractor={(item) => item.chat_id}
+            />
+          ) : (
+            <LoadingSpinner style={styles.loadingSpinner} />
```
Comment (rickyrombo) [inline packages/mobile/src/screens/chat-screen/ChatListScreen.tsx:53]: will have to think about error state too

File: packages/mobile/src/screens/chat-screen/ChatScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  useEffect(() => {
+    dispatch(fetchMoreMessages({ chatId }))
+  }, [dispatch, chatId])
+
+  const otherUser = useSelector((state) => getOtherChatUsers(state, chatId))
+
+  return (
+    <Screen
+      url={url}
+      // variant='secondary'
```
Comment (rickyrombo) [inline packages/mobile/src/screens/chat-screen/ChatScreen.tsx:117]: comment

File: packages/mobile/src/screens/chat-screen/ChatScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    width: spacing(5),
+    height: spacing(5),
+    fill: palette.primary
+  }
+}))
+
+/**
+ * Checks to see if the message was sent within the time threshold for grouping it with the next message
+ */
+const hasTail = (message: ChatMessage, newMessage?: ChatMessage) => {
```
Comment (rickyrombo) [inline packages/mobile/src/screens/chat-screen/ChatScreen.tsx:82]: might be able to move this to common too?

File: packages/mobile/src/screens/chat-screen/ChatScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+} from '@audius/common'
+import type { ChatMessage } from '@audius/sdk'
+import dayjs from 'dayjs'
+import { View } from 'react-native'
+import { useDispatch, useSelector } from 'react-redux'
+
+import IconSend from 'app/assets/images/iconSend.svg'
+import { TextInput, Screen, FlatList, ScreenContent } from 'app/components/core'
+import LoadingSpinner from 'app/components/loading-spinner'
+// import { UserBadges } from 'app/components/user-badges'
```
Comment (rickyrombo) [inline packages/mobile/src/screens/chat-screen/ChatScreen.tsx:18]: comment

**Review by rickyrombo — APPROVED** [review]

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-PAY-882

**Pull request merged**